packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -0,0 +1,125 @@
+        new Error(nls.localize('error_retrieving_previous_authorizations'))
+      );
+    }
+    const authInfoObjects = Promise.all(
+      authFilesArray.map(fileName => {
+        const filePath = path.join(os.homedir(), '.sfdx', fileName);
+        return readFileAsync(filePath, 'utf8')
+          .then(fileData => JSON.parse(fileData))
+          .catch(err => null);
+      })
+    );
+    return authInfoObjects;
+  }
+
+  public async filterAuthInfo(authInfoObjects: FileInfo[]) {
+    authInfoObjects = authInfoObjects.filter(fileData =>
+      isNullOrUndefined(fileData.scratchAdminUsername)
+    );
+    authInfoObjects = authInfoObjects.filter(fileData =>
+      isNullOrUndefined(fileData.isDevHub)
+    );
+    const authUsernames = authInfoObjects.map(file => file.username);
+    const aliases = await Aliases.create(Aliases.getDefaultOptions());
+    const authList = [];
+    for (const username of authUsernames) {
+      const alias = await aliases.getKeysByValue(username);
+      if (alias.length > 0) {
+        authList.push(alias + ' - ' + username);
+      } else {
+        authList.push(username);
+      }
+    }
+    return authList;
+  }
+}
+
+let statusBarItem: StatusBarItem;
+
+export async function updateOrgList() {
+  const orgList = new OrgList();
+  const authInfoObjects = await orgList.getAuthInfoObjects();
+  const authUsernameList = await orgList.filterAuthInfo(authInfoObjects);
+  return authUsernameList;
+}
+
+export async function setDefaultOrg(): Promise<
+  CancelResponse | ContinueResponse<{}>
+> {
+  const orgList = await updateOrgList();
+  orgList.unshift(
+    nls.localize('force_auth_web_login_authorize_org_text'),
+    nls.localize('force_org_create_default_scratch_org_text')
+  );
+  const selection = await vscode.window.showQuickPick(orgList);
+  if (!selection) {
+    return { type: 'CANCEL' };
+  }
+
+  if (selection === nls.localize('force_auth_web_login_authorize_org_text')) {
+    vscode.commands.executeCommand('sfdx.force.auth.web.login');
+    return {
+      type: 'CONTINUE',
+      data: {}
+    };
+  } else if (
+    selection === nls.localize('force_org_create_default_scratch_org_text')
+  ) {
+    vscode.commands.executeCommand('sfdx.force.org.create');
+    return { type: 'CONTINUE', data: {} };
+  } else {
+    const usernameOrAlias = selection.split(' ', 1);
+    vscode.commands.executeCommand('sfdx.force.config.set', usernameOrAlias);
+    return { type: 'CONTINUE', data: {} };
+  }
+}
+
+export async function showOrg() {
+  if (!statusBarItem) {
+    statusBarItem = window.createStatusBarItem(StatusBarAlignment.Left, 50);
+    statusBarItem.command = 'sfdx.force.set.default.org';
+    statusBarItem.show();
+  }
+  const editor = window.activeTextEditor;
+  if (!editor) {
+    statusBarItem.hide();
+    return;
+  }
+  displayDefaultUsername();
+}
+
+export async function displayDefaultUsername() {
+  const defaultUsernameorAlias = await getDefaultUsernameOrAlias();
```

Comment:
This method will return the global `defaultusername` when one cannot be found for the current project. Because of this, there is a situation in which the user has run `SFDX: Log Out from All Authorized Orgs` so there are no auth files available but the `sfdx-config.json` file does not remove/update the global 'defaultusername.' In that case, the user will see a default in the status bar, but when they click on it they will receive an error. If we'd rather that not happen, I can also change it to only refer to the local config file for the `defaultusername` info.